test/test_ncbi_taxa.py

```diff
@@ -1537,87 +1525,268 @@ def test_validate_index_higher_taxa(
+                }
+            ),
+            (
+                (INFO, "Loading NCBITaxa worksheet"),
+                (INFO, "Reading NCBI taxa data"),
+                (INFO, "3 NCBI rank fields found:"),
+                (INFO, "Validating row 1: 562"),
+                (ERROR, "Worksheet name is not a string: 562"),
+                (INFO, "Match found for 562"),
+                (INFO, "Indexing taxonomic hierarchy"),
+                (INFO, "Added superkingdom Bacteria"),
+                (INFO, "Added phylum Pseudomonadota"),
+                (INFO, "Added class Gammaproteobacteria"),
+                (INFO, "Added order Enterobacterales"),
+                (INFO, "Added family Enterobacteriaceae"),
+                (INFO, "Added genus Escherichia"),
+                (INFO, "NCBITaxa contains 1 errors"),
+            ),
+            id="numeric name",
+        ),
+        pytest.param(
+            DotMap(
+                {
+                    "data_columns": [
+                        ("E coli ",),
+                        ("Enterobacteriaceae",),
+                        ("Escherichia",),
+                        ("coli",),
+                    ],
+                    "headers": ["name", "family", "genus", "species"],
+                }
+            ),
+            (
+                (INFO, "Loading NCBITaxa worksheet"),
+                (INFO, "Reading NCBI taxa data"),
+                (INFO, "3 NCBI rank fields found:"),
+                (INFO, "Validating row 1:"),
+                (ERROR, "Worksheet name has whitespace padding: 'E coli '"),
+                (INFO, "Match found for E coli"),
+                (INFO, "Indexing taxonomic hierarchy"),
+                (INFO, "Added superkingdom Bacteria"),
+                (INFO, "Added phylum Pseudomonadota"),
+                (INFO, "Added class Gammaproteobacteria"),
+                (INFO, "Added order Enterobacterales"),
+                (INFO, "Added family Enterobacteriaceae"),
+                (INFO, "Added genus Escherichia"),
+                (INFO, "NCBITaxa contains 1 errors"),
+            ),
+            id="padded name",
+        ),
+        pytest.param(
+            DotMap(
+                {
+                    "data_columns": [
+                        ("E coli",),
+                        (123,),
+                        ("Escherichia",),
+                        ("coli",),
+                    ],
+                    "headers": ["name", "family", "genus", "species"],
+                }
+            ),
+            (
+                (INFO, "Loading NCBITaxa worksheet"),
+                (INFO, "Reading NCBI taxa data"),
+                (INFO, "3 NCBI rank fields found:"),
+                (INFO, "Validating row 1:"),
+                (ERROR, "Rank family has non-string or empty string value: 123"),
+                (INFO, "NCBITaxa contains 1 errors"),
+            ),
+            id="non string taxon",
+        ),
```

Comment:
Guess a useful test here is whether if you give the full outdated hierarchy for `Cytophaga marina` (i.e. including Cytophaga as the genus) whether it passes or fails

Reply:
Or probably not if we are using time stamping